fix: refacto

immortal-tofu opened the pull request:
**Separation**
The first goal is to separate relayer logic (send inputs to relayer, send user decrypt request to relayer) to core logic (encrypt inputs, generate private/public keys, generate EIP712).

**Tests**
Rewrite tests for encryption which suffered of multiple rewrite (could be better, but it's a first step)

0xalexbel commented:
🧐 May be we should export all the types the user may encounter: 
- `RelayerEncryptedInput` (ok)
- `EIP712` (from keypair)
- `CtHandleContractPairParam` (listed in HTTPZInstance type definition)
- `PublicParams` (listed in HTTPZInstance type definition)

jatZama commented on src/index.ts:
```diff
@@ -25,7 +34,7 @@ export type HTTPZInstance = {
     contractAddress: string,
     userAddress: string,
-  ) => ZKInput;
+  ) => RelayerEncryptedInput;
```
why did you rename ZKInput to RelayerEncryptedInput? same input could be sent directly by the user to gateway

immortal-tofu replied:
`RelayerEncryptedInput` is the object whch allows you to interact with the realyer when you call `encrypt()` while `EncryptedInput` is the SDK's object which "just" encrypt the data.

jatZama commented:
If I understood correctly major difference now is that you have two functions for inputs, createEncryptedInput and createEncryptedRelayerInput, where relayerUrl is required only for second one.

0xalexbel commented:
While you are working on cleaning stuff: 
- maybe you could rename test `encrypt/decrypt` into `encrypt` in encrypt.test.ts ?

@httpz/sdk/sdk/* is now internal ? right ?

immortal-tofu commented:
> If I understood correctly major difference now is that you have two functions for inputs, `createEncryptedInput` and `createRelayerEncryptedInput`, where relayerUrl is required only for second one. Did I miss other major changes?

Yes you're right. And also I moved all relayer related stuff in the relayer directory.

Otherwise, I also fixed encryption test where fetchMock was not used correctly and we've implemented weird cache to handle that.

jatZama commented on src/relayer/sendEncryption.ts:
```diff
@@ -0,0 +1,151 @@
+
+export type RelayerEncryptedInput = Omit<EncryptedInput, 'encrypt'> & {
+  _input: EncryptedInput;
+  encrypt: () => Promise<{
```
since there is already an encrypt() function inside the _input value, don't you think some other more explicit name here is better than another encrypt() function? Maybe something like "encryptAndSendToRelayer" ?

immortal-tofu replied:
User wouldn't see the `encrypt` function from SDK, so I think there is no confusion here for them

jatZama replied:
ok

0xalexbel commented:
The only type name that is questionable to me is `CtHandleContractPairParam` which looks probably great for an expert, but incomprehensible for a newbie (furthermore this type looks internal when you read its name, and it is not self explanatory). Having said all of that, I don't have a proposal... maybe `HandleContractPairs` or `HttpzHandleContractPairs` (a handle is always Ct). Maybe we could give a global name for bytes32 handles in js/ts like `HTTPZHandle` or `HttpzHandle` or something like this. This could even be defined

immortal-tofu commented:
> The only type name that is questionable to me is `CtHandleContractPairParam` which looks probably great for an expert, but incomprehensible for a newbie (furthermore this type looks internal when you read its name, and it is not self explanatory). Having said all of that, I don't have a proposal... maybe `HandleContractPairs` or `HttpzHandleContractPairs` (a handle is always Ct). Maybe we could give a global name for bytes32 handles in js/ts like `HTTPZHandle` or `HttpzHandle` or something like this. This could even be defined

Done. I just don't createa `HTTPZHandle` because you can't defined the type except alias "string", so it doesn't make much sense imo.
